src/journal/journald-server.c

```diff
@@ -2038,3 +2042,35 @@ int config_parse_line_max(
 
         return 0;
 }
+
+int config_parse_compress(const char* unit,
+                          const char *filename,
+                          unsigned line,
+                          const char *section,
+                          unsigned section_line,
+                          const char *lvalue,
+                          int ltype,
+                          const char *rvalue,
+                          void *data,
+                          void *userdata) {
+        JournalCompressOptions* compress = data;
+        int r;
+
+        if (streq(rvalue, "1")) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0,
+                           "Compress= ambiguously specified as 1, enabling compression with default threshold");
+                compress->enabled = true;
+        } else if (streq(rvalue, "0")) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0,
+                           "Compress= ambiguously specified as 0, disabling compression");
+                compress->enabled = false;
+        } else if ((r = parse_boolean(rvalue)) >= 0)
+                compress->enabled = r;
+        else if (parse_size(rvalue, 1024, &compress->threshold_bytes) == 0)
+                compress->enabled = true;
+        else if (!streq(rvalue, ""))
+                /* Ignore empty values and just use the default */
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse Compress= value, ignoring: %s", rvalue);
```

Comment:
nah, this is not sufficient. The idea of permitting assignment of the empty string is that subsequent lines can cancel the effect of previous lines. Think:

```
Compress=4711
Compress=815
Compress=
```

The second line should override the effect of the first line, but the third line should then undo the effect of both, and reset to the built-in default.

THis behaviour is useful, as we generally permit .d/ drop-in files for all our configuration files, where it is very handy to cancel out the effect of previous drop-ins.

Reply:
TIL. Will re-spin